src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,34 +1,56 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import { SortBy } from '../../types/SortBy';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+type FilterTodo = {
+  onSelect: (value: SortBy) => void,
+  onInput: (value: string) => void,
+  query: string,
+  onCross: (value: string) => void,
+};
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button
-          data-cy="clearSearchButton"
-          type="button"
-          className="delete"
+export const TodoFilter: React.FC<FilterTodo> = ({
+  onSelect,
+  onInput,
+  query,
+  onCross,
+}) => {
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            onChange={(e) => onSelect(e.target.value as unknown as SortBy)}
+            data-cy="statusSelect"
+          >
+            <option value={SortBy.all}>All</option>
+            <option value={SortBy.active}>Active</option>
+            <option value={SortBy.completed}>Completed</option>
+          </select>
+        </span>
+      </p>
+
+      <p className="control is-expanded has-icons-left has-icons-right">
+        <input
+          data-cy="searchInput"
+          type="text"
+          className="input"
+          placeholder="Search..."
+          value={query}
+          onChange={(e) => onInput(e.target.value)}
         />
-      </span>
-    </p>
-  </form>
-);
+        <span className="icon is-left">
+          <i className="fas fa-magnifying-glass" />
+        </span>
+
+        <span className="icon is-right" style={{ pointerEvents: 'all' }}>
+          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+          <button
+            data-cy="clearSearchButton"
+            type="button"
+            className="delete"
+            onClick={() => onCross(query)}
```

Comment:
```suggestion
            onClick={onCross}
```
the `onCross` (should be renamed) doesn't accept any argument, why do you pass query there